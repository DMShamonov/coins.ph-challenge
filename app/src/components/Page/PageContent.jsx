// @flow

import * as React from 'react';
import classnames from 'classnames';

import type { PageContentPropsType } from './types.js.flow';

export default function PageContent({
  className,
  children,
}: PageContentPropsType): React.Node {
  return (
    <div className={classnames('page__content', className)}>
      {children}
    </div>
  );
}
