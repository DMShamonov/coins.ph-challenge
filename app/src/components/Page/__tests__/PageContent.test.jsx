import React from 'react';
import { shallow } from 'enzyme';

import PageContent from '../PageContent';

describe('`PageContent` component should render correctly', () => {
  test('with default props', () => {
    expect(shallow(<PageContent>__CONTENT__</PageContent>)).toMatchSnapshot();
  });

  test('with `className` prop', () => {
    expect(shallow(<PageContent className="__CLASSNAME__">__CONTENT__</PageContent>)).toMatchSnapshot();
  });
});
