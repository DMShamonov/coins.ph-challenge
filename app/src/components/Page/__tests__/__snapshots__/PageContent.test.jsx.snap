// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`PageContent\` component should render correctly with \`className\` prop 1`] = `
<div
  className="page__content __CLASSNAME__"
>
  __CONTENT__
</div>
`;

exports[`\`PageContent\` component should render correctly with default props 1`] = `
<div
  className="page__content"
>
  __CONTENT__
</div>
`;
